client/src/components/container/portal/LogInPage/index.js

```diff
@@ -0,0 +1,81 @@
+import React, { Component } from 'react'
+import { Form, Input, Button, Spin, Alert, message } from 'antd';
+import axios from 'axios';
+
+import './style.css'
+export class PortalLogin extends Component {
+  state = {
+    isLoade: false,
+    pinCode: '',
+    serverError: '',
+    msg: '',
+    error: false,
+  }
+
+  handleChange = ({target:{value}}) => {
+    this.setState({ pinCode: value, error: false });
+  };
+
+  onFinish = ({ pinCode }) => {
+    this.setState({ isLoade: true })
+    axios.post('/api/v1/portal/login', { pinCode}).then(({ data }) => {
+      if (data.status === 301) {
+        this.props.history.push('/portal/attendance');
+        message.success(data.msg, 10);
+      } else if (data.status === 400) {
+        this.setState({
+          error: true,
+          msg: data.msg,
+          pinCode: '',
+        })
+      } else if (data.status === 401) {
+        this.setState({
+          error: true,
+          msg: data.msg,
+          pinCode: '',
+        })
+      }
```

Comment:
all cases of error will be handled at the same way here so you don't need to check for every case.
```suggestion
      } else {
        this.setState({
          error: true,
          msg: data.msg,
          pinCode: '',
        })
      }
```